cpp/arcticdb/python/normalization_checks.cpp

```diff
@@ -153,7 +152,6 @@ bool check_pandas_like(bool is_append,
                                     stop);
                 }
 
-                normalization::check<ErrorCode::E_UPDATE_NOT_SUPPORTED>(is_append, "update() on ROWCOUNT index is not implemented");
```

Comment:
Why is it OK to drop this check?

Reply:
This check is not really meaningful because:
1. it is checking only the normalization data, so before my changes, if there is no normalization data, the appends would still go through. So the check is not really correct. And now we have cases with normalization data that previous passed but are now failing, which doesn't seem correct.
2. I have explicitly added tests that test appends/updates with ROWCOUNT and they seems to be working:
https://github.com/man-group/ArcticDB/blob/b085ec11202324644ee4141b8d201d269739e35e/python/tests/integration/arcticdb/version_store/test_update_with_date_range.py#L140